internal/provider/resource_aws_transit_gateway_attachment_test.go

```diff
@@ -0,0 +1,231 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var (
+	// using unique names for AWS resource to make debugging easier
+	tgwAttUniqueAWSName        = fmt.Sprintf("hcp-tf-provider-test-%d", rand.Intn(99999))
+	testAccTGWAttachmentConfig = fmt.Sprintf(`
+provider "aws" {
+  region = "us-west-2"
+}
+
+resource "hcp_hvn" "main" {
+  hvn_id         = "main-hvn"
```

Comment:
For consistency and also I got a conflict since this id is used in a lot of our examples
```suggestion
  hvn_id         = "test-hvn"
```